cfg/rationalize.go

```diff
@@ -14,10 +14,28 @@
 
 package cfg
 
+import "net/url"
+
+func decodeURL(u string) (string, error) {
+	decodedUrl, err := url.Parse(u)
```

Comment:
nit: Feel free to ignore - Can we add a TODO to replace `Parse` call with [ParseRequestURI](https://pkg.go.dev/net/url#ParseRequestURI).